File: backend/services/endpoint_service.go
package services

import (
	iDao "github.com/pdftts/webapp/daos/interfaces"
	"github.com/pdftts/webapp/models"
)

type EndpointService struct {
	dao iDao.EndpointDaoInterface
}

func NewEndpointService(dao iDao.EndpointDaoInterface) *EndpointService {
	return &EndpointService{dao: dao}
}

func (service *EndpointService) GetAllEndpoints() ([]models.Endpoint, error) {
	return service.dao.GetAllEndpointsDao()
}

File: frontend/src/api/services/endpoint.service.ts
import { Endpoint } from "../../models/endpoint";
import axios from "../axios";

export const getAllEndpoints = async (): Promise<Endpoint[]> => {
    try {
        const response = await axios.get(`/endpoint`);
        return response.data as Endpoint[];
    } catch (error) {
        console.error('Error fetching endpoints:', error);
        throw error;
    }
}